src/securejoin.rs

```diff
@@ -510,8 +525,23 @@ pub(crate) async fn handle_securejoin_handshake(
             context.emit_event(EventType::ContactsChanged(Some(contact_id)));
             if let Some(joining_chat_id) = joining_chat_id {
                 // Join group.
-                chat::add_contact_to_chat_ex(context, Nosync, joining_chat_id, contact_id, true)
-                    .await?;
+                let res = chat::add_contact_to_chat_ex(
+                    context,
+                    Nosync,
+                    joining_chat_id,
+                    contact_id,
+                    true,
+                )
+                .await;
+                if let Err(e) = res {
```

Comment:
This part can now be reverted, solution for dealing with exceptional receive_imf errors is at https://github.com/chatmail/core/pull/7401